validator/sawtooth_validator/consensus/proxy.py

```diff
@@ -147,7 +147,16 @@ def forks(self):
     # Using blockstore and state database
     def blocks_get(self, block_ids):
         '''Returns a list of blocks.'''
-        return self._get_blocks(block_ids)
+        try:
+            return self._get_blocks(block_ids)
+        except UnknownBlock:
+            peers = self._gossip.get_peers()
+            peered_connections = [conn_id for conn_id in peers
+                                  if self._gossip.get_connection_status(conn_id) == PeerStatus.PEER]
+            for peer_id in peered_connections:
+                for block_id in block_ids:
+                    self._gossip.send_block_request(block_id.hex(), peer_id)
+            raise UnknownBlock()
```

Comment:
Add description for the error or if it not required then propagate from `except UnknownBlock as e`